model/Aktivita/Aktivita.php

```diff
@@ -1945,12 +1946,16 @@ public function zkontrolujZdaSeMuzePrihlasit(
             }
         }
 
-        // přihlášení na navázané aktivity (jen pokud není teamleader)
-        if ($this->a['dite'] && $this->pocetPrihlasenych() > 0) {
+        // přihlášení na navázané aktivity
+        if ($this->a['dite']) {
             $deti = $this->deti();
             if (count($deti) === 1) {
-                current($deti)->prihlas($uzivatel, $prihlasujici, self::STAV | ($parametry & self::UKAZAT_DETAILY_CHYBY));
-            } else {
+                try {
+                    current($deti)->prihlas($uzivatel, $prihlasujici, self::STAV | ($parametry & self::UKAZAT_DETAILY_CHYBY));
+                } catch (Exception $e) {
```

Comment:
Zobrazovat běžným uživatelům detail Exception je nehezké a může to být i bezpečnostní díra (bůhví která exception vyskočila a co měla v message).

Podle použití `\Gamecon\Aktivita\Aktivita::UKAZAT_DETAILY_CHYBY` by `\Gamecon\Aktivita\Aktivita::prihlas` měla vyhazovat speciální exception `\Chyba`, která je výslovně určená pro BFU a která, když není zachycená, se změní na flash message.
Proto bych tu zachytil dva typy errorů, Chyba a Throwable (Throwable je ještě obecnější než Exception a zahrnuje i systémové errory jako chybný datový typ atd.) 
```php
} catch (Chyba $chyba) {
    throw new \Chyba('Nepodařilo se přihlásit na navazující aktivitu s chybou: ' . $chyba->getMessage());
} catch (\Throwable $throwable) {
    throw new \Chyba('Nepodařilo se přihlásit na navazující aktivitu. Interní chyba systému.');
   Vyjimkovac::vytvorZGlobals()->zaloguj($throwable);
}
```
poznámka: vytvářet logger uvnitř entity je sice ošklivé, ale metoda `Aktivita::prihlas` by měla být spíše v controlleru, nebo v service a tam by byl dostupný jako dependency, takže vlastně OK.